plugins/inputs/cloudwatch/cloudwatch.go

```diff
@@ -367,66 +366,37 @@ func (c *CloudWatch) getDataQueries(filteredMetrics []filteredMetric) map[string
-				dataQueries[*metric.Namespace] = append(dataQueries[*metric.Namespace], types.MetricDataQuery{
-					Id:    aws.String("maximum_" + id),
-					Label: aws.String(snakeCase(*metric.MetricName + "_maximum")),
-					MetricStat: &types.MetricStat{
-						Metric: &filtered.metrics[j],
-						Period: aws.Int32(int32(time.Duration(c.Period).Seconds())),
-						Stat:   aws.String(StatisticMaximum),
-					},
-				})
+			var accountID *string
+			if c.IncludeLinkedAccounts {
+				accountID = aws.String(filtered.accounts[j])
+				(*dimension)["account"] = filtered.accounts[j]
 			}
-			if filtered.statFilter.Match("minimum") {
-				c.queryDimensions["minimum_"+id] = dimension
-				dataQueries[*metric.Namespace] = append(dataQueries[*metric.Namespace], types.MetricDataQuery{
-					Id:    aws.String("minimum_" + id),
-					Label: aws.String(snakeCase(*metric.MetricName + "_minimum")),
-					MetricStat: &types.MetricStat{
-						Metric: &filtered.metrics[j],
-						Period: aws.Int32(int32(time.Duration(c.Period).Seconds())),
-						Stat:   aws.String(StatisticMinimum),
-					},
-				})
-			}
-			if filtered.statFilter.Match("sum") {
-				c.queryDimensions["sum_"+id] = dimension
-				dataQueries[*metric.Namespace] = append(dataQueries[*metric.Namespace], types.MetricDataQuery{
-					Id:    aws.String("sum_" + id),
-					Label: aws.String(snakeCase(*metric.MetricName + "_sum")),
-					MetricStat: &types.MetricStat{
-						Metric: &filtered.metrics[j],
-						Period: aws.Int32(int32(time.Duration(c.Period).Seconds())),
-						Stat:   aws.String(StatisticSum),
-					},
-				})
+
+			statisticTypes := map[string]string{
+				"average":      "Average",
+				"maximum":      "Maximum",
+				"minimum":      "Minimum",
+				"sum":          "Sum",
+				"sample_count": "SampleCount",
 			}
-			if filtered.statFilter.Match("sample_count") {
-				c.queryDimensions["sample_count_"+id] = dimension
-				dataQueries[*metric.Namespace] = append(dataQueries[*metric.Namespace], types.MetricDataQuery{
-					Id:    aws.String("sample_count_" + id),
-					Label: aws.String(snakeCase(*metric.MetricName + "_sample_count")),
-					MetricStat: &types.MetricStat{
-						Metric: &filtered.metrics[j],
-						Period: aws.Int32(int32(time.Duration(c.Period).Seconds())),
-						Stat:   aws.String(StatisticSampleCount),
-					},
-				})
+
+			for statisticType, statistic := range statisticTypes {
+				if filtered.statFilter.Match(statisticType) {
+					queryID := statisticType + "_" + id
+					c.queryDimensions[queryID] = dimension
+					dataQueries[*metric.Namespace] = append(dataQueries[*metric.Namespace], types.MetricDataQuery{
+						Id:        aws.String(queryID),
+						AccountId: accountID,
+						Label:     aws.String(snakeCase(*metric.MetricName + "_" + statisticType)),
+						MetricStat: &types.MetricStat{
+							Metric: &filtered.metrics[j],
+							Period: aws.Int32(int32(time.Duration(c.Period).Seconds())),
+							Stat:   aws.String(statistic),
+						},
+					})
+				}
```

Comment:
Can we please save one level of indentation via
```suggestion
				if !filtered.statFilter.Match(statisticType) {
					continue
				}
				queryID := statisticType + "_" + id
				c.queryDimensions[queryID] = dimension
				dataQueries[*metric.Namespace] = append(dataQueries[*metric.Namespace], types.MetricDataQuery{
					Id:        aws.String(queryID),
					AccountId: accountID,
					Label:     aws.String(snakeCase(*metric.MetricName + "_" + statisticType)),
					MetricStat: &types.MetricStat{
						Metric: &filtered.metrics[j],
						Period: aws.Int32(int32(time.Duration(c.Period).Seconds())),
						Stat:   aws.String(statistic),
					},
				})
```

Reply:
feat: adding support for cloudwatch cross account observability. reindenting the code